enumerable.rb

```diff
@@ -0,0 +1,152 @@
+# rubocop:disable Style/For
+
+# rubocop: disable Metrics/CyclomaticComplexity
+
+# rubocop:disable Metrics/PerceivedComplexity
+
+# rubocop: disable Metrics/ModuleLength
+
+# rubocop: disable Style/ConditionalAssignment
+
+# Enumerable module
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    index = 0
+    for i in self
+      yield i, index
+      index += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    my_each { |x| array << x if yield x }
+
+    array
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |x| return false if yield(x) == false }
+      return true
+    elsif arg.is_a?(Class)
+      my_each { |x| return false unless x.is_a?(arg) }
+    elsif arg.nil?
+      my_each { |i| return false if i.nil? || i == false }
+    elsif arg.is_a?(Regexp)
+      my_each { |x| return false unless x.match(arg) }
+    else
+      my_each { |i| return false if i != arg }
+    end
+    true
+  end
+
+  def my_any?(arg = true)
+    if block_given?
+      my_each { |x| return true if yield(x) == true }
+      return false
+    elsif arg.is_a?(Class)
+      my_each { |x| return true if x.is_a?(arg) }
+    elsif arg.nil?
+      my_each { |x| return true if x.nil? || x == false }
+    elsif arg.is_a?(Regexp)
+      my_each { |x| return true if x.match(arg) }
+    elsif arg.is_a?(String)
+      my_each { |x| return true if x == arg }
+    else
+      my_each { |x| return true if x == true }
+    end
+    false
+  end
+
+  def my_none?(arg = false)
```

Comment:
# my_none?

- [ ] when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern:

```ruby
words = %w[dog door rod blade]
words.my_none?(5) === words.none?(5) => false
```

Reply:
I apply it on my project it returned true I even try it on my friend's projects it also returned true then I thought maybe all our projects even the one that approved is not working sight so I apply it on the reviewer's projects also returned true so I really don't know what to do about it I looked a lot of examples but I couldn't find anything to help me